.github/actions/download.sh

```diff
@@ -10,16 +10,36 @@
 # - use download-all target to get all files when package has multiple (arch specific) files.
 
 set -o pipefail
+set -e  # Exit on any command failure
+
+# Function to download a package and handle errors
+download_package() {
+    local package_dir="$1"
+    local target="$2"
+    
+    echo "===> Attempting to download: ${package_dir}"
+    if ! make -C "${package_dir}" "${target}"; then
+        echo "Error: Failed to download ${package_dir}. Exiting." >&2
+        exit 1
+    fi
+    echo "===> Successfully downloaded: ${package_dir}"
+
+    # Verify checksum after download
+    echo "===> Verifying checksum for: ${package_dir}"
+    if ! make -C "${package_dir}" checksum; then
+        echo "Error: Checksum verification failed for ${package_dir}. Exiting." >&2
+        exit 1
+    fi
+    echo "===> Checksum verified: ${package_dir}"
+}
```

Comment:
one success message is enough (IMHO)

```suggestion
    fi
    # Verify checksum after download
    if ! make -C "${package_dir}" checksum; then
        echo "Error: Checksum verification failed for ${package_dir}. Exiting." >&2
        exit 1
    fi
    echo "===> Successfully downloaded: ${package_dir}"
}
```